### DISCUSSION TITLE
Error 0x80073FCA when updating or uninstalling packaged apps on Windows 10

### DISCUSSION THREAD

**Discussion by RDMacLachlan** [microsoft/WindowsAppSDK#4913] [Announcements]
## Overview: 
Our analysis into Windows and client logging has determined that this problem may be preventing users from updating or uninstalling packaged apps on Window 10 devices. We've isolated a recent servicing update for the WinAppSDK (ver. 1.6.2) which introduced an issue with the deployment stack processing that resulted in this impact. 

## Resolution:
The following resolutions have been tried out by the community and partners. Select one of the following options to resolve a device in a problematic state: 
* Install the [KB5046714](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fsupport.microsoft.com%2Fhelp%2F5046714&data=05%7C02%7CRoy.MacLachlan%40microsoft.com%7C50177fd2fa3144a4210408dd0a436c57%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638678006926544442%7CUnknown%7CTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7C0%7C%7C%7C&sdata=OLmRY06kie5rUORpaElPg93ZCO%2FHAcRvPnpJ4rE0GjM%3D&reserved=0) Microsoft Update.
* Install the new WinAppSDK 1.6.3 three times (matching to your device architecture)
	* [Installer (x64)](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Faka.ms%2Fwindowsappsdk%2F1.6%2F1.6.241114003%2Fwindowsappruntimeinstall-x64.exe&data=05%7C02%7CRoy.MacLachlan%40microsoft.com%7C3c0a153aa65544f0b4cb08dd09c110e3%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638677447068563375%7CUnknown%7CTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7C0%7C%7C%7C&sdata=BjCyaDvGHOj84z51HsJIWK3Hie0sU9QEblyEHNXvKJI%3D&reserved=0) (Most common)
	* [Installer (arm64)](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Faka.ms%2Fwindowsappsdk%2F1.6%2F1.6.241114003%2Fwindowsappruntimeinstall-arm64.exe&data=05%7C02%7CRoy.MacLachlan%40microsoft.com%7C3c0a153aa65544f0b4cb08dd09c110e3%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638677447068610989%7CUnknown%7CTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7C0%7C%7C%7C&sdata=2iZzgR6lZIG%2FNGnfPZkPA3K722KNUzTHIiuBTynrGuc%3D&reserved=0)
 
## Identify if your device was impacted
The following two conditions must be true to be impacted by this issue:
* Windows 10 operating system
* WinAppSDK 1.6.2 installed
	* The following PowerShell command will return a value if you have the WinAppSDK 1.6.2 installed on your device:
```PowerShell 
Get-Appxpackage *WindowsAppRuntime.1.6* -AllUsers | Where { $_.Version -eq '6000.311.13.0' }
```

## Impacted User Experience:
Users attempting to update or uninstall a packaged app on a Windows 10 device will receive either of the following error Message: “Deployment failed with HRESULT: 0x80073CFA”

* Uninstall or update error showing error code 0x80073CFA:
![image](https://github.com/user-attachments/assets/1541bc2f-df67-4fa6-bda2-18d88165c10e)
 
* Attempting to uninstall an app in the Start menu does not remove the app from the list.
* The Microsoft Store shows a “Something happened on our end” error with a Retry button:
![image](https://github.com/user-attachments/assets/1854e474-fccd-4cea-b297-ab17f43a4c30)

 
* Deployment failed with HRESULT: 0x80073CFA:
![image](https://github.com/user-attachments/assets/f729b1f9-1c26-423c-a522-ae3a98504733)

**Comment by King-Taz** [comment]
I'm a Win 10 user who updated to Win 11 to [fix this issue](https://github.com/microsoft/WindowsAppSDK/issues/4881#issuecomment-2494273067). The change to Win 11 solved the problem about uninstalling local apps but created a new issue in VS. I cannot deploy Maui app using net9, the Deploy checkbox in configuration is grayed out.

![image](https://github.com/user-attachments/assets/e63d5a30-febb-4f6c-af32-cd7301c7e295)

The only work around I've found is to start the Maui app using net8 and manually update it to net9 through the project file. How can this issue be repaired?

**Reply by manada02** [reply]
Same I have 2 apps.exe wher I delted their projects but apps remains. I updated for a new windows update 1 of 2  these appa that I couldnt uninstall werer gone but that other app still show error 0x80073cfa

**Reply by haonanttt** [reply]
Hi @King-Taz , Based on my investigation of the issue you encountered, it seems to be related to a .NET 9 feature and not the Windows App SDK:

According to [MAUI public doc](https://learn.microsoft.com/en-us/dotnet/maui/windows/setup?view=net-maui-9.0#convert-an-unpackaged-net-maui-windows-app-to-packaged), there’s a new checkbox to determine whether your app is packaged or unpackaged:
![image](https://github.com/user-attachments/assets/10a34ffc-9728-4ac6-bbf0-3c7bbbe4f688)
When you select the `Create a Windows MSIX package` checkbox, the 'Deploy' checkbox in the configuration manager becomes clickable.

However, here is 1 more thing I want to mention: If your concern about the checkbox is due to behavior in .NET 8—where an unselected 'Deploy' checkbox prevented debugging—I’ve observed in my repro that even if the 'Deploy' checkbox is unclickable in a new MAUI app using .NET 9, you can still debug the app successfully. As a result, I didn’t notice any significant business impact from this issue.

If you have additional concerns or believe this still impacts your workflow, please consider opening an issue with the MAUI team. Thanks!

**Comment by jmwoods301** [comment]
"Install the new WinAppSDK" worked well for us.

Not comfortable installing a Preview Update to fix the issue, so thanks to Mike Crider for a clean solution.

**Reply by King-Taz** [reply]
> "Install the new WinAppSDK" worked well for us.


That doesn't work, doesn't change anything. It is intended for Win10 platform, not Win11.

**Reply by jmwoods301** [reply]
As mentioned in the first thread post...

The following two conditions must be true to be impacted by this issue:

    Windows 10 operating system
    WinAppSDK 1.6.2 installed 

Plus the thread title...

Error 0x80073FCA when updating or uninstalling packaged apps on Windows 10

So yes, it is intended for Win 10...and it works.

Roll back to Windows 10, or restore from a disk image, and run it.

**Reply by King-Taz** [reply]
> Roll back to Windows 10, or restore from a disk image, and run it.

lot of work for a temp fix that has issues

**Comment by AT-4** [comment]
I've installed the new WinAppSDK, but I'm still encountering the issue with the Microsoft store, not sure how to proceed.

Win10, KB5046714 & WinAppSDK 1.6.3 installed.

**Reply by yeelam-gordon** [reply]
Thanks for reporting this, here is the steps to troubleshoot:

1. Ensure that WinAppSDK 1.6.3 is installed.
Run the following PowerShell command (as Administrator) to check:
```powershell
Get-AppxPackage *WindowsAppRuntime.1.6* -AllUsers | Where { $_.Version -eq '6000.318.2304.0' }
```
If this version is not installed, try the following command to install it (it may fail once or twice, so please retry if needed):
```powershell
winget install Microsoft.WindowsAppRuntime.1.6
```

2. Retry App Update/Install in Microsoft Store
Click "Retry" on the app update/install in the Microsoft Store.
If it fails again, open Windows Event Viewer (type eventvwr in the Run dialog) and check the Application Logs:
Event Viewer > Windows Logs > Application
Look for recent entries (current minute) with the following details:
Level: Error
Source: Application Error
Event ID: 1000
Check if the error details reference appxdeploymentserver.dll.

**Reply by AT-4** [reply]
> Thanks for reporting this, here is the steps to troubleshoot:
> 
> 1. Ensure that WinAppSDK 1.6.3 is installed.
>    Run the following PowerShell command (as Administrator) to check:
> 
> ```powershell
> Get-AppxPackage *WindowsAppRuntime.1.6* -AllUsers | Where { $_.Version -eq '6000.318.2304.0' }
> ```
> 
> If this version is not installed, try the following command to install it (it may fail once or twice, so please retry if needed):
> 
> ```powershell
> winget install Microsoft.WindowsAppRuntime.1.6
> ```
> 
> 2. Retry App Update/Install in Microsoft Store
>    Click "Retry" on the app update/install in the Microsoft Store.
>    If it fails again, open Windows Event Viewer (type eventvwr in the Run dialog) and check the Application Logs:
>    Event Viewer > Windows Logs > Application
>    Look for recent entries (current minute) with the following details:
>    Level: Error
>    Source: Application Error
>    Event ID: 1000
>    Check if the error details reference appxdeploymentserver.dll.



Checked if it was installed through Powershell, the versions didn't match. Installed with the command provided, worked the first try. 

After this, attempted to install an app through the Microsoft Store, same issue occurs. 

Went to check Event Viewer, (Windows Logs>Application), not seeing any recent entries with the Error level.

**Reply by yeelam-gordon** [reply]
Then it looks like a different issue on the Microsoft Store itself.
Try to install(or upgrade/uninstall) any simple app by winget, it should be successful.

**Reply by AT-4** [reply]
> Then it looks like a different issue on the Microsoft Store itself. Try to install(or upgrade/uninstall) any simple app by winget, it should be successful.

Still getting the same issue (& error code) through winget.
![image](https://github.com/user-attachments/assets/40798558-92f0-488a-a0e3-4d91ef44c687)

**Reply by AT-4** [reply]
Still having issues, but other apps are downloading/updating just fine. Nothing has changed

![image](https://github.com/user-attachments/assets/cfdc92eb-20ac-4f45-a752-f693ce83859d)

**Reply by AT-4** [reply]
Nothing has changed, still having the same issue.

**Comment by manada02** [comment]
same event Viewegh doesn't show nothing.

Získat Outlook pro Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by ghost1372** [comment]
i am on Window 11 And I wasn't affected by the Windows 10 issues, but I don't know why, but I ran the commands mentioned in [this comment](https://github.com/microsoft/WindowsAppSDK/issues/4881#issuecomment-2480939942) on Windows 11 and now I have problems.
Every time I create a new project or open an existing project, I have to manually deploy the project first and then run it.
Otherwise, I will encounter the error "The project is not registered and deploy it."
The second problem is that hot reload does not work and the changes are not visible and I have to restart the program every time.
I repaired Visual Studio but it didn't work.
Is there a solution to get rid of this problem?
